src/api/socket.c

```diff
@@ -367,6 +392,29 @@ static void *socket_connection_handler_thread(void *socket_desc)
 	sprintf(threadname, "socket-%i", sock);
 	prctl(PR_SET_NAME, threadname, 0, 0, 0);
 
+	// Ensure this thread can be canceled at any time (not only at
+	// cancellation points)
+	pthread_setcanceltype(PTHREAD_CANCEL_ASYNCHRONOUS, NULL);
+
+	// Store TID of this thread
+	lock_shm();
+	unsigned int tid;
+	for(tid = 0; tid < MAX_API_THREADS; tid++)
+	{
+		if(api_threads[tid] == 0)
+		{
+			api_threads[tid] = pthread_self();
+			api_tids[tid] = gettid();
+			break;
+		}
+	}
+	unlock_shm();
+	if(tid == MAX_API_THREADS)
+	{
+		logg("Not able to spawn new API thread, limit reached.");
```

Comment:
Can this include the value of `MAX_API_THREADS` instead of just reporting that the limit was reached? 

Something like "`limit of n threads reached.`"